### PR TITLE
Add support for regenerating protocol buffers, and refresh .pb.go files

### PR DESCRIPTION
**Pull request by euanh**
Previously the build container did not provide `protoc` or `protoc-gen-gogo`;  now these are available in the container.   There is also a make pattern rule to regenerate `.pb.go` files from `.proto` files.

I regenerated the `.pb.go` files.   `agent.go.pb` was regenerated recently and does not change, however older `.pb.go` files change slightly when regenerated with newer tools.

### CODE REVIEW

**Comment by euanh** [comment]
@fcrisciani, @ctelfer: This should make regenerating protocol buffers more consistent.    I'll try to come up with a nice `regenerate-protobufs` target in the Makefile as well.

**Comment by euanh** [comment]
I haven't tried building moby with these changes yet.   Is there a convenient subset of E2E tests which particularly exercises libnetwork?

**Comment by euanh** [comment]
I tried using the vendored version of gogoproto, but `protoc` generated bad import statements which `go build` couldn't resolve:
```
import _ "gogoproto"
```
instead of 
```
import _ "github.com/gogo/protobuf/gogoproto"
```

File: Makefile [hunk 1]
```diff
@@ -123,3 +123,14 @@ gosimple: ## run gosimple
...
 shell: builder
 	@${docker} ${SHELL}
+
+# Rebuild protocol buffers.
+# These may need to be rebuilt after vendoring updates, so .pb.go files are declared .PHONY so they are always rebuilt.
+PROTO_FILES=$(shell find . -path ./vendor -prune -o -name \*.proto -print)
+PB_FILES=$(PROTO_FILES:.proto=.pb.go)
+
+%.pb.go: %.proto
+	${docker} protoc -I=. -I=/go/src -I=/go/src/github.com/gogo/protobuf -I=/go/src/github.com/gogo/protobuf/protobuf --gogo_out=./ $<
```
Comment (fcrisciani) [inline Makefile]: don't not have to be in this PR, but would be great to fail if the PR while compiled by the CI has a diff with the file generated compared to the files coming from the PR itself. This way we can enforce that the PR contains all the proper information.
Also does it makes sense to have build target depend on the protobuf generation?

**Comment by selansen** [comment]
LGTM

**Comment by euanh** [comment]
@fcrisciani @selansen @ctelfer I'm currently running the tests that @ctelfer suggested and tracking down a couple of failures.

**Comment by euanh** [comment]
I reproduced the same failures running a mixed-version cluster with moby master (without my changes) on the leader and 17.06 on two other nodes.   Updating all nodes to moby master made them pass.   Similarly, with my changes only on the leader I had a couple of failures; with my changes on all nodes the tests passed.

**Pull request merged**